core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/PaginationFilter.java

```diff
@@ -54,6 +58,24 @@ public static PaginationFilter boxUse(long id) {
     };
   }
 
+  public static <M extends Identifiable, D> List<D> bulkSearch(Collection<String> names, PaginatedDataSource<M> service,
+      Function<M, D> dto) {
+    return names.stream()//
+        .filter(name -> !LimsUtils.isStringBlankOrNull(name))//
+        .flatMap(WhineyFunction.flatRethrow(name -> {
+          Collection<M> matches = service.list(0, 0, true, "id", exactQuery(name));
+          if (matches.isEmpty()) {
+            throw new IllegalArgumentException(String.format("Cannot find \"%s\".", name));
```

Comment:
I'm not terribly fond of the `Error: Internal Server Error` that gets thrown when the user requests something that doesn't exist. It makes it sound more dire than it actually is. Can we remove the `Internal Server Error` part?